File: CSharp/PackageLockDependency.cs
using System.Collections.Generic;

namespace VerifyNodeModules
{
    public class PackageLockDependency
    {
        public string Version { get; set; }
        public Dictionary<string, PackageLockDependency> Dependencies { get; set; } = new Dictionary<string, PackageLockDependency>();
    }
}

File: FSharp/PackageLockDependency.fs
namespace VerifyNodeModules

type PackageLockDependency(version: string, dependencies: Map<string, PackageLockDependency>) =

    let coalesce x replacement =
        if not (obj.ReferenceEquals(x, null)) then
            x
        else
            replacement

    member this.Version = version

    member this.Dependencies =
        coalesce dependencies Map.empty
